Method lookup on returned interface

The following program `sample.go` triggers a panic:

```go
package main
    
import "fmt"
    
type I interface {
    A() string
    B() string
}       
    
type s struct{}
    
func NewS() (I, error) {
    return &s{}, nil
}           
        
func (c *s) A() string { return "a" }
func (c *s) B() string { return "b" }

func main() {
    s, _ := NewS()
    fmt.Println(s.A())
}
```

Expected result:
```console
$ go run ./sample.go
a
```

Got:
```console
$ yaegi ./sample.go
sample.go:20:10: panic
runtime error: invalid memory address or nil pointer dereference
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc00039a000, 0x40)
        /home/marc/sdk/go1.14.2/src/runtime/debug/stack.go:24 +0x9d
github.com/containous/yaegi/interp.(*Interpreter).Eval.func1(0xc0001c9e10)
        /home/marc/dev/yaegi/interp/interp.go:320 +0xbc
panic(0xce2960, 0x1548e50)
        /home/marc/sdk/go1.14.2/src/runtime/panic.go:969 +0x166
github.com/containous/yaegi/interp.runCfg.func1(0xc0003700a0, 0xc000385400)
        /home/marc/dev/yaegi/interp/run.go:113 +0x20c
panic(0xce2960, 0x1548e50)
        /home/marc/sdk/go1.14.2/src/runtime/panic.go:969 +0x166
github.com/containous/yaegi/interp.getMethodByName.func1(0xc0003700a0, 0xc000042940)
        /home/marc/dev/yaegi/interp/run.go:1303 +0x1d7
github.com/containous/yaegi/interp.runCfg(0xc000385400, 0xc0003700a0)
        /home/marc/dev/yaegi/interp/run.go:119 +0x6d
github.com/containous/yaegi/interp.(*Interpreter).run(0xc000001500, 0xc000384b00, 0xc000370000)
        /home/marc/dev/yaegi/interp/run.go:97 +0x239
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc000001500, 0xc00037a000, 0x11b, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/marc/dev/yaegi/interp/interp.go:388 +0x41f
main.main()
        /home/marc/dev/yaegi/cmd/yaegi/yaegi.go:94 +0x548

```

--- a/_test/interface37.go
+++ b/_test/interface37.go
@@ -1,0 +1,23 @@
+package main
+
+type I interface {
+	A() string
+	B() string
+}
+
+type s struct{}
+
+func NewS() (I, error) {
+	return &s{}, nil
+}
+
+func (c *s) A() string { return "a" }
+func (c *s) B() string { return "b" }
+
+func main() {
+	s, _ := NewS()
+	println(s.A())
+}
+
+// Output:
+// a

--- a/interp/run.go
+++ b/interp/run.go
@@ -678,7 +678,7 @@ func call(n *node) {
 			switch {
 			case c.ident == "_":
 				// Skip assigning return value to blank var.
-			case c.typ.cat == interfaceT:
+			case c.typ.cat == interfaceT && rtypes[i].cat != interfaceT:
 				rvalues[i] = genValueInterfaceValue(c)
 			default:
 				rvalues[i] = genValue(c)

